test/lit/passes/gto-removals.wast

```diff
@@ -1495,3 +1495,27 @@
     )
   )
 )
+
+;; The type $A is public because it is on an exported global. As a result we
+;; cannot remove the unused i32 field from its child or grandchild.
+(module
+  ;; CHECK:      (type $A (sub (struct (field (mut i32)))))
+  (type $A (sub (struct (field (mut i32)))))
```

Comment:
Should we also add a test where the field is not present on `$A`, so it can still be removed from `$B` and `$C`?

Reply:
Good catch, this was actually a bug. The loop on supers here would need to also handle the super not having the field, mirroring the code elsewhere.

Overall this seems to make the assertion more and more complex, past the point of being worth it, so I removed it. We do still cover this in depth in tests.